### PR TITLE
[Routing] Change RouteCollection:getUri()

### PR DESCRIPTION
**Pull request by crouillon**
- use the request values if available
- introduce bbapp.routing.default_protocol in container parameters
- add ApplicationInterface::isClientSAPI() (CR?)

### CODE REVIEW

File: Routing/RouteCollection.php [hunk 1]
```diff
@@ -51,66 +54,132 @@ class RouteCollection extends sfRouteCollection implements DumpableServiceInterf
...
 
-        if (true === $container->hasParameter('bbapp.routing.resource_uri_prefix')) {
-            $this->uri_prefixes[self::RESOURCE_URL] = $container->getParameter('bbapp.routing.resource_uri_prefix');
+        if ($container->hasParameter('bbapp.routing.resource_uri_prefix')) {
+            $this->uriPrefixes[self::RESOURCE_URL] = $container->getParameter('bbapp.routing.resource_uri_prefix');
         }
 
-        $this->is_restored = false;
+        if ($container->hasParameter('bbapp.routing.default_protocol')) {
+            $this->defaultScheme = $container->getParameter('bbapp.routing.default_protocol');
```
Comment (mickaelandrieu) [inline Routing/RouteCollection.php]: :+1:

**Comment by eric-chau** [comment]
:+1: nice improvements @crouillon =)

**Comment by mickaelandrieu** [comment]
@eric-chau merge it \o/

**Comment by ndufreche** [comment]
:+1: cool

**Pull request merged**